roles/openshift_master/templates/master.yaml.v1.j2

```diff
@@ -15,8 +15,8 @@ assetConfig:
 {% if 'logging_public_url' in openshift.master %}
   loggingPublicURL: {{ openshift.master.logging_public_url }}
 {% endif %}
-{% if 'metrics_public_url' in openshift.master %}
-  metricsPublicURL: {{ openshift.master.metrics_public_url }}
+{% if openshift_hosted_metrics_deploy | default(false) | bool %}
```

Comment:
As it stands here it's always going to be set after openshift_master_facts role runs, which loses us the ability to tell if it was set by user or defaulted. I did post a suggestion for the other PR though that would build on top of this to let us detect here and write this if deploy = true, *or* this var was set in inventory. Proposing this as a followup.